api/v1beta1/clusterclass_types.go

```diff
@@ -100,6 +100,26 @@ type ControlPlaneClass struct {
 	// referenced above is Machine based and supports setting replicas.
 	// +optional
 	MachineHealthCheck *MachineHealthCheckClass `json:"machineHealthCheck,omitempty"`
+
+	// NodeDrainTimeout is the total amount of time that the controller will spend on draining a node.
+	// The default value is 0, meaning that the node can be drained without any time limitations.
+	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
+	// NOTE: This value can be overridden while defining a Cluster.Topology using this ControlPlaneClass.
```

Comment:
```suggestion
	// NOTE: This value can be overridden while defining a Cluster.Topology.
```
nit: I would drop this part as users can't pick the "ControlPlaneClass"

(same below)